### PR TITLE
New module MetaPhlAn4

### LINKED ISSUE #3153: new module: MetaPhlAn 4
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
## PR checklist

Closes #3153  

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/nf-core/metaphlan/metaphlan/main.nf [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { UNTAR                 } from '../../../../../modules/nf-core/untar/main.nf'
```
Comment (sofstam) [inline tests/modules/nf-core/metaphlan/metaphlan/main.nf:5]:
```suggestion
include { UNTAR               } from '../../../../../modules/nf-core/untar/main.nf'
```

**Review by sofstam — CHANGES_REQUESTED** [review]

File: tests/modules/nf-core/metaphlan/metaphlan/main.nf (hunk 1 above)
Comment (LilyAnderssonLee) [reply tests/modules/nf-core/metaphlan/metaphlan/main.nf:5]: UNTAR was used to extract compressed database tar.gz file. So I think it's necessary to include it here.

Comment (sofstam) [reply tests/modules/nf-core/metaphlan/metaphlan/main.nf:5]: This was a format suggestion, not removing the whole line :)

Comment (LilyAnderssonLee) [reply tests/modules/nf-core/metaphlan/metaphlan/main.nf:5]: Sorry. I didn't check carefully. Thanks Sofia!

File: tests/modules/nf-core/metaphlan/metaphlan/nextflow.config
[the remark this answers is not in the document]
Comment (LilyAnderssonLee) [reply tests/modules/nf-core/metaphlan/metaphlan/nextflow.config:1]: According to the MetaPhlAn4 manual, there are a few different types of input formats, which is why I have included different processes similar to the MetaPhlAn3 module.

[the remark this answers is not in the document]
Comment (sofstam) [reply tests/modules/nf-core/metaphlan/metaphlan/nextflow.config:1]: I only see one process in metaphlan3.

https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/metaphlan3/metaphlan3/nextflow.config

[the remark this answers is not in the document]
Comment (LilyAnderssonLee) [reply tests/modules/nf-core/metaphlan/metaphlan/nextflow.config:1]: Yes, you are right that there is only one process in `MetaPhlAn3`. I have now combined the processes. 
It is necessary to use the `SAMTOOLS_VIEW` process to produce the correct format, which is '`sam`', and can be used as input for the `MetaPhlAn3` process.

**Review by sofstam — APPROVED** [review]
I think you can merge.

**Pull request merged**